core/src/main/scala/tamer/Setup.scala

```diff
@@ -1,13 +1,24 @@
 package tamer
 
+import zio._
+import zio.blocking.Blocking
+import zio.clock.Clock
 import zio.kafka.serde.Serializer
 
-case class Setup[-K, -V, S](
-    serdes: Setup.Serdes[K, V, S],
-    defaultState: S,
-    tamerStateKafkaRecordKey: Int,
-    repr: String = "no repr string implemented, if you want a neat description of the source configuration please implement it"
-)
+abstract class Setup[-R, K, V, S] {
+  val serdes: Setup.Serdes[K, V, S]
+  val defaultState: S
+  val stateKey: Int
+  val recordKey: (S, V) => K
+  val repr: String = "no repr string implemented, if you want a neat description of the source configuration please implement it"
+  def iteration(currentState: S, queue: Queue[Chunk[(K, V)]]): RIO[R, S]
+
+  final val run: ZIO[R with Has[KafkaConfig] with Blocking with Clock, TamerError, Unit] = runLoop.provideSomeLayer(Tamer.live(this))
+  final def runWith[E >: TamerError, R1 <: Has[_]](layer: ZLayer[ZEnv, E, R1])(
+      implicit ev: ZEnv with R1 <:< R with Has[KafkaConfig] with Blocking with Clock,
+      tagged: Tag[R1]
+  ): ZIO[ZEnv, E, Unit] = run.provideCustomLayer(layer)
```

Comment:
All frontends must `extend Setup[R, K, V, S]` and provide all the above implementations.

They access Tamer from their specialized `*Setup` by invoking either of these (and maybe we could add others that take pure values, ZIO, etc.